packages/core/src/test/awsService/ec2/remoteSessionManager.test.ts

```diff
@@ -0,0 +1,56 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import assert from 'assert'
+import * as sinon from 'sinon'
+import { Ec2RemoteSessionManager } from '../../../awsService/ec2/remoteSessionManager'
+import { SsmClient } from '../../../shared/clients/ssmClient'
+
+describe('Ec2RemoteSessionManager', async function () {
```

Comment:
is there a session manager for codecatalyst dev envs? if so, it may be worth generalizing it. All of this new test code suggests that it's worth having 1 generalized concept instead of several specific ones.

Reply:
For CodeCatalyst, we start the connection within the connect script here https://github.com/aws/aws-toolkit-vscode/blob/0164d4145e58ae036ddf3815455ea12a159d491d/packages/core/resources/codecatalyst_connect#L97 so we don't have the resulting `SessionId` in memory. Also, I am unsure if the same problem of sessions not terminating still happens in CodeCatalyst due to some of these implementation differences.